solana/program_v2/programs/gateway_v2/src/util.rs

```diff
@@ -23,3 +26,101 @@ pub trait OnChainSizeWithArg<Arg> {
     /// Gets the size with an argument
     fn on_chain_size_with_arg(arg: Arg) -> usize;
 }
+
+pub fn get_gatekeeper_fees(fees: &[GatekeeperFees], mint: Pubkey) -> &GatekeeperFees {
+    fees.iter().find(|&&x| x.token == mint).unwrap()
+}
+
+pub fn get_network_fees(fees: &[NetworkFees], mint: Pubkey) -> &NetworkFees {
+    fees.iter().find(|&&x| x.token == mint).unwrap()
+}
+
+/// get_fees
+/// Returns two fees in the correct unit
+/// First result returns the fee for the guardian
+/// Second result returns the gatekeeper fee
+pub fn work_out_fees(fee: u64, split: u16) -> (u64, u64) {
+    let percentage = (split as f64).div(100_f64);
+    let guardian_fee = (fee as f64).mul(percentage);
```

Comment:
Let's call this `network_fee` ? the idea of `gaurdian` doesn't really exist outside of tests/implementations... 

I'd think of `gaurdian fee` as a collection of `network fees` (as it's 1:n from gaurdian to network)

Reply:
That makes sense, will update now.

Reply:
Done!